code/modules/clothing/masks/_masks.dm

```diff
@@ -68,33 +67,14 @@
 		M.update_worn_mask()
 
 //Proc that moves gas/breath masks out of the way, disabling them and allowing pill/food consumption
-/obj/item/clothing/mask/proc/adjustmask(mob/living/carbon/user)
-	if(user?.incapacitated())
-		return
-	mask_adjusted = !mask_adjusted
-	if(!mask_adjusted)
-		icon_state = initial(icon_state)
-		clothing_flags |= visor_flags
-		flags_inv |= visor_flags_inv
-		flags_cover |= visor_flags_cover
-		to_chat(user, span_notice("You push \the [src] back into place."))
-		slot_flags = initial(slot_flags)
-	else
-		icon_state += "_up"
-		to_chat(user, span_notice("You push \the [src] out of the way."))
-		clothing_flags &= ~visor_flags
-		flags_inv &= ~visor_flags_inv
-		flags_cover &= ~visor_flags_cover
+/obj/item/clothing/mask/visor_toggling(mob/living/user)
+	. = ..()
+	icon_state = "[initial(icon_state)][up ? "_up" : ""]" //generic behavior, if you add a mask that doesnt do this then remove it and manually add it to subtypes
```

Comment:
Any reason why this can't be in `update_icon_state`?

Reply:
thats fair i just copied that behavior from theold one